DirectProgramming/C++SYCL_FPGA/Tutorials/Features/optimization_levels/minimum_latency/src/minimum_latency.cpp

```diff
@@ -0,0 +1,141 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include "exception_handler.hpp"
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+#include <vector>
+
+using namespace sycl;
+
+class Kernel;
+
+#if defined(FPGA_SIMULATOR)
+constexpr int kInputSize = 10;
+#else
+constexpr int kInputSize = 1000;
+#endif
+
+typedef int RGBType;
+typedef std::vector<RGBType> RGBVec;
+typedef float GreyType;
+typedef std::vector<GreyType> GreyVec;
+
+// Return the execution time of the event, in seconds
+double GetExecutionTime(const event &e) {
+  double start_k = e.get_profiling_info<info::event_profiling::command_start>();
+  double end_k = e.get_profiling_info<info::event_profiling::command_end>();
+  double kernel_time = (end_k - start_k) * 1e-9; // ns to s
+  return kernel_time;
+}
+
+GreyType Compute(RGBType r, RGBType g, RGBType b) {
+  GreyType output =
+      (GreyType)r * 0.3f + (GreyType)g * 0.59f + (GreyType)b * 0.11f;
+  return output;
+}
+
+void RunKernel(const RGBVec &r, const RGBVec &g, const RGBVec &b,
+               GreyVec &out) {
+#if defined(FPGA_EMULATOR)
+  ext::intel::fpga_emulator_selector selector;
```

Comment:
You need to update your selectors to the latest ones

Reply:
Note that you'll now need -DFPGA_HARDWARE when running FPGA compiles to select the correct selector